### DISCUSSION TITLE
Turn on or off?

### DISCUSSION THREAD

**Discussion by Fr3d3rick83** [windkh/node-red-contrib-shelly#102] [Q&A] [answered]
Hi, I connected a shelly gen. 1 node to my shelly but how can I turn that shelly on?

**Comment by windkh** [comment]
Just have a look at the examples

**Reply by Fr3d3rick83** [reply]
I did, but it wont turn off or on the shelly.
<img width="563" alt="Bildschirmfoto 2022-11-02 um 17 16 22" src="https://user-images.githubusercontent.com/97901424/199543452-d9bfa18c-bb4a-4d21-98db-a02940f407a0.png">

**Reply by windkh** [reply]
Send me your flow please.
What shelly do you use?

**Reply by Fr3d3rick83** [reply]
This is the flow:
Its a Shelly Plug.

[
    {
        "id": "681f72c95d42565c",
        "type": "shelly-gen1",
        "z": "8be703368a859619",
        "hostname": "192.168.178.56",
        "description": "Steckdose leah",
        "mode": "polling",
        "server": "",
        "outputmode": "event",
        "uploadretryinterval": 5000,
        "pollinginterval": 5000,
        "pollstatus": false,
        "getstatusoncommand": true,
        "devicetype": "Relay",
        "outputs": 1,
        "x": 460,
        "y": 1380,
        "wires": [
            [
                "13cc189725b8a272"
            ]
        ]
    },
    {
        "id": "13cc189725b8a272",
        "type": "debug",
        "z": "8be703368a859619",
        "name": "",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "true",
        "targetType": "full",
        "statusVal": "",
        "statusType": "auto",
        "x": 650,
        "y": 1380,
        "wires": []
    },
    {
        "id": "6b22877f50c05ef8",
        "type": "inject",
        "z": "8be703368a859619",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "x": 240,
        "y": 1380,
        "wires": [
            [
                "681f72c95d42565c"
            ]
        ]
    },
    {
        "id": "9e1714a60659b8d4",
        "type": "vsh-virtual-device",
        "z": "8be703368a859619",
        "name": "Turn Shelly on",
        "topic": "Turn Shelly on",
        "connection": "93040f1af5c0d6dd",
        "template": "SWITCH",
        "retrievable": false,
        "passthrough": false,
        "diff": false,
        "filter": false,
        "filterTopic": false,
        "x": 220,
        "y": 1320,
        "wires": [
            [
                "681f72c95d42565c"
            ]
        ]
    },
    {
        "id": "d71c8b1b4349c2c0",
        "type": "vsh-virtual-device",
        "z": "8be703368a859619",
        "name": "Turn Shelly off",
        "topic": "Turn Shelly off",
        "connection": "93040f1af5c0d6dd",
        "template": "SWITCH",
        "retrievable": false,
        "passthrough": false,
        "diff": false,
        "filter": false,
        "filterTopic": false,
        "x": 220,
        "y": 1280,
        "wires": [
            [
                "681f72c95d42565c"
            ]
        ]
    },
    {
        "id": "93040f1af5c0d6dd",
        "type": "vsh-connection",
        "name": "f_ludeck@yahoo.de",
        "port": "8883",
        "accessTokenExpiry": "1667410493412",
        "debug": false,
        "backendUrl": "https://kfd5m4a21f.execute-api.eu-west-1.amazonaws.com/dev",
        "lwaClientId": "amzn1.application-oa2-client.3f1bb07133854b078261ad43f2484c18"
    }
]

**Comment by windkh** [comment]
@Fr3d3rick83 I assume that you are not familiar with node-red yet?
Well, the node needs the correct msg.payload to act correctly.

Please read the manual:
https://github.com/windkh/node-red-contrib-shelly#relays-shelly-1-2-1pm-1l-plug-plugs-uni

Here is a sample flow where you can see how a switch works
(see also https://github.com/windkh/node-red-contrib-shelly/blob/master/examples/switch.json)

[{"id":"2d791f81ae6dfa2c","type":"debug","z":"b1246db37f9d2f38","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","x":730,"y":300,"wires":[]},{"id":"ae7595966f77f53a","type":"function","z":"b1246db37f9d2f38","name":"0 on","func":"msg.payload = {\n    relay:0,\n    on:true\n};\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":230,"y":280,"wires":[["b1ddcf9f17392251"]]},{"id":"19b506322a9c08e7","type":"function","z":"b1246db37f9d2f38","name":"0 off","func":"msg.payload = {\n    relay:0, \n    on:false\n};\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":230,"y":320,"wires":[["b1ddcf9f17392251"]]},{"id":"53a377c30f3b4de8","type":"inject","z":"b1246db37f9d2f38","name":"Turn On","repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":80,"y":280,"wires":[["ae7595966f77f53a"]]},{"id":"c4914622404b741f","type":"inject","z":"b1246db37f9d2f38","name":"Turn Off","repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":90,"y":320,"wires":[["19b506322a9c08e7"]]},{"id":"d89e5b66446a7e72","type":"inject","z":"b1246db37f9d2f38","name":"Poll Status","repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":80,"y":220,"wires":[["b1ddcf9f17392251"]]},{"id":"0bf0ad05d6e49371","type":"function","z":"b1246db37f9d2f38","name":"0 toggle","func":"msg.payload = {\n    relay:0, \n    turn:\"toggle\"\n};\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":240,"y":360,"wires":[["b1ddcf9f17392251"]]},{"id":"ae8d0da801ed9715","type":"inject","z":"b1246db37f9d2f38","name":"Toggle","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":90,"y":360,"wires":[["0bf0ad05d6e49371"]]},{"id":"b1ddcf9f17392251","type":"shelly-gen1","z":"b1246db37f9d2f38","hostname":"192.168.178.189","description":"","mode":"polling","server":"033a6e9acf499057","uploadretryinterval":"5000","pollinginterval":5000,"pollstatus":false,"getstatusoncommand":true,"devicetype":"Relay","outputs":1,"credentials":{"username":"","password":""},"x":530,"y":300,"wires":[["2d791f81ae6dfa2c"]]},{"id":"033a6e9acf499057","type":"shelly-gen1-server","port":"20001","hostname":""}]

**Reply by Fr3d3rick83** [reply]
I´ll getr an error: 
SyntaxError: JSON.parse: expected ',' or '}' after property value in object at line 1 column 1489 of the JSON data.

0, \n turn:"toggle"\n};\

**Comment by windkh** [comment]
You can directly import the examples from the sidebar-->Import:

![image](https://user-images.githubusercontent.com/10910767/200073613-76b29b68-d619-40a0-93a3-4f731d161809.png)

**Comment by windkh** [comment, marked as answer]
@Fr3d3rick83 any success?

**Reply by Fr3d3rick83** [reply]
Yes, I didnt know, that I always need a function.
